app/controllers/api/v1/mattermost_controller.rb

```diff
@@ -0,0 +1,25 @@
+class Api::V1::MattermostController < ApplicationController
+  before_action :authenticate_mattermost
+
+  def create
+    if params[:command]
+      render json: Commands.run(command_params).response
+    else
+      head :unauthorized
+    end
+  end
+
+  private
+
+  def authenticate_mattermost
+    auth_present = params[:token] && params[:team_domain]
+    valid_token = params[:token] == ENV['MATTERMOST_TOKEN']
```

Comment:
For equality checks involving an authentication token, [ActiveSupport::SecurityUtils.secure_compare](https://apidock.com/rails/ActiveSupport/SecurityUtils/secure_compare/class) should be used.